book/scheduling-and-threading.md

```diff
@@ -327,116 +275,129 @@ situation---some devices have more [CPU cores][cores] than others, or are more
 sensitive to battery power usage.
 :::
 
+[cores]: https://en.wikipedia.org/wiki/Multi-core_processor
+
 Long-lived threads
 ==================
 
-Python not only lets you create timers, but you can create and manipulate
-threads themselves, via the `threading.Thread` class. Let's use threads to
-implement async `XHMLHttpRequest`, a feature we left out of
-[Chapter 10](security.md#cross-site-requests). At the time, we implemented it
-as a synchronous API, but in fact, the synchronous version of that API is almost
-useless for real websites,^[It's also a huge performance footgun, for the same
-reason we've been adding async tasks in this chapter!] because the whole point
-of using this API is to keep the website responsive to the user while
-network requests are going on.
-
-Our approach will be to start a thread, run some code on it that does the
-request and gets a response, then schedule a `Task` to send the response back
-to the script. Starting a thread is easy: define a function that is the "main"
-function of the thread, then start the thread with the function passed as the
-`target` argument. When the main function exits, the thread will automatically
-die.
-
-In this example:
-``` {.python expected=False}
-def run:
```

Comment:
This example is too complicated, you think?

Reply:
Yeah. I think there's also a bit of a question here of defining our audience. If they haven't seen threads at all before, will they be able to follow the description here? It also introduces tricky questions like whether the console out is thread-safe, which we don't actually need to raise.